components/Form/FormInput.js

```diff
@@ -0,0 +1,53 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+
+const specialChars = /[`!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?~]/;
+
+// const regex = /[a-zA-Z0-9 .]+/;
+
+export const FormInput = ({
+  defaultValue,
+  onChange,
+  labelClassName,
+  label,
+  required,
+  value,
+  pattern,
+  className,
+  ...props
+}) => {
+  const [userInput, setUserInput] = useState(defaultValue);
+  const [isValid, setIsValid] = useState(true);
+
+  const inputHandler = (e) => {
+    if (onChange) {
+      onChange(e);
+    } else {
+      pattern && setIsValid(!specialChars.test(e.target.value));
```

Comment:
I prefer to use `if`, not this inline condition, that does assign value to nothing (some eslint config would not allow that)

Consider doing classic `if(pattern) { }` approach